vncviewer/Viewport.cxx

```diff
@@ -606,6 +606,11 @@ int Viewport::handle(int event)
     if (Fl::event_button3())
       buttonMask |= 4;
 
+    if(Fl::event_buttons()&FL_BUTTON(4))
+        buttonMask |= 128;
+    if(Fl::event_buttons()&FL_BUTTON(5))
+        buttonMask |= 256;
```

Comment:
Are you sure about these? FLTK follows X11's button numbering. So I would expect 7 and 8 here.

(this also means we should probably pass through every high button from FLTK, to support all extra buttons)

Reply:
Yeah, i tested and use these values. I did find it odd that java and c had different button combos (the clients). I'll look into this again after extending all buttons.

Reply:
I had a chance to look into Java's behavior in the context of implementing horizontal scroll wheel events.  Java Buttons 4 through 7 are assigned differently depending on the O/S, because horizontal scroll wheel events are implemented differently in different O/S's.  On macOS and Windows, horizontal scroll wheel events are implemented as vertical scroll wheel events with the Shift key held down.  Since Java has a separate event subclass for scroll wheel events, it doesn't need to use Java Buttons 4 and 5 for scroll wheel events.  Thus, on macOS and Windows, Java Buttons 4 and 5 are used for the front and back mouse buttons.  On X11, however, horizontal scroll wheel events are implemented using X11 Buttons 6 and 7.  Java/X11 still uses a separate event subclass for vertical scroll wheel events, but it uses Java Buttons 4 and 5 for horizontal scroll wheel events, which means that it uses Java Buttons 6 and 7 for the front and back mouse buttons.